runtime/perfetto/db.go

```diff
@@ -95,16 +95,22 @@ type DB struct {
 	replicaNumCache *lru.Cache[replicaCacheKey, int]
 }
 
-// Open opens the default trace database on the local machine.
-func Open(ctx context.Context) (*DB, error) {
+// DatabaseFilePath returns the path of the file that stores the trace database.
+func DatabaseFilePath(mode string) (string, error) {
 	dataDir, err := files.DefaultDataDir()
 	if err != nil {
-		return nil, err
+		return "", err
 	}
+	return filepath.Join(dataDir, fmt.Sprintf("perfetto_%s.db", mode)), nil
+}
 
-	// TODO(mwhittaker): Use a different db for every deployer. Have the purge
-	// commands delete the db.
-	fname := filepath.Join(dataDir, "perfetto.db")
+// Open opens the default trace database on the local machine. mode specifies
+// the weaver mode (e.g. single, multi, ssh) for which the database is being open.
+func Open(ctx context.Context, mode string) (*DB, error) {
```

Comment:
> Done. Although I am not happy with the strings constants I sprinkled around. It would've helped if there was a type for specifying mode of weaver tool but I didn't want to make that change in context of this CL. Let me know if you have any suggestions.

This looks great to me! An alternative would be to change `Open` to receive the filename of the database: 

```go
func Open(ctx context.Context, filename string) (*DB, error)  
```

and then delete `DatabaseFilePath`. Then, every deployer's package (i.e. single, multi, ssh) defines a var or function that returns the filename of their perfetto database. `DashboardSpec` can either take this filename or the `perfetto.DB` directly.

I like your approach and this approach, both, so don't feel compelled to make any changes. I think we'll eventually want to make a bigger change later anyway to avoid having to purge everything all the time.

Reply:
Thanks for the suggestion. Let me stick with what I have currently and we can make appropriate changes as we improve our tool versioning story in the future.